tokio/src/runtime/task/join.rs

```diff
@@ -252,7 +252,7 @@ impl<T> JoinHandle<T> {
     }
 
     /// Returns a new `AbortHandle` that can be used to remotely abort this task.
-    pub(crate) fn abort_handle(&self) -> super::AbortHandle {
+    pub fn abort_handle(&self) -> super::AbortHandle {
         self.raw.ref_inc();
```

Comment:
If we are making this public, then I would prefer the documentation to be more comprehensive. For example, it would be nice to say that you can still abort via the JoinHandle, and that if you abort via the AbortHandle, then the JoinHandle will complete with an aborted error, or if the future completes first, its output.

Reply:
Good idea, I have adapted the docs of `JoinHandle::abort` for this.